.teamcity/src/subprojects/build/core/ProjectCore.kt

```diff
@@ -9,27 +9,97 @@ import jetbrains.buildServer.configs.kotlin.v2019_2.triggers.*
 
 data class JDKEntry(val name: String, val env: String)
 data class OSEntry(val name: String, val agentString: String)
+data class BrowserEntry(val name: String, val dockerContainer: String)
 
 val operatingSystems = listOf(OSEntry("macOS", "Mac OS X"), OSEntry("Linux", "Linux"), OSEntry("Windows", "Windows"))
 val jdkVersions = listOf(JDKEntry("Java 8","JDK_18"), JDKEntry("Java 11", "JDK_11"))
+val browsers = listOf(BrowserEntry("Chrome", "stl5/ktor-test-image:latest"))
 
 object ProjectCore : Project({
     id("ProjectKtorCore")
     name = "Core"
     description = "Ktor Core Framework"
     for (os in operatingSystems) {
         for (jdk in jdkVersions) {
-            buildType(BuildTemplate(os, jdk))
+            buildType(CoreBuild(os, jdk))
         }
     }
+    for (os in operatingSystems) {
+        buildType(NativeBuild(os))
+    }
+    for (browser in browsers) {
+        buildType(JavaScriptBuild(browser))
+    }
+})
+
+class JavaScriptBuild(val browserEntry: BrowserEntry): BuildType({
+    id("KtorMatrixJavaScript_${browserEntry.name}".toExtId())
+    name = "JavaScript on ${browserEntry.name}"
 
+    vcs {
+        root(VCSCore)
+    }
+    triggers {
+        vcs {
+            quietPeriodMode = VcsTrigger.QuietPeriodMode.USE_DEFAULT
+            branchFilter = ""
+        }
+    }
+    steps {
+        gradle {
+            name = "Parallel assemble"
+            tasks = "assemble"
```

Comment:
Maybe we should use an explicit `--parallel` switch here

Reply:
Not sure what you mean? This is meant to be parallel.